src/Modules/StatCan.OrchardCore.AjaxForms/Controllers/AjaxFormController.cs

```diff
@@ -0,0 +1,156 @@
+        private readonly IScriptingManager _scriptingManager;
+        private readonly IWorkflowManager _workflowManager;
+
+        public AjaxFormController(
+            ILogger<AjaxFormController> logger,
+            IContentManager contentManager,
+            IContentDefinitionManager contentDefinitionManager,
+            IContentItemDisplayManager contentItemDisplayManager,
+            IUpdateModelAccessor updateModelAccessor,
+            IScriptingManager scriptingManager,
+            IWorkflowManager workflowManager = null
+            )
+        {
+            _logger = logger;
+            _contentManager = contentManager;
+            _contentDefinitionManager = contentDefinitionManager;
+            _contentItemDisplayManager = contentItemDisplayManager;
+            _updateModelAccessor = updateModelAccessor;
+            _scriptingManager = scriptingManager;
+            _workflowManager = workflowManager;
+        }
+
+        [ValidateAntiForgeryToken]
+        [HttpPost]
+        public async Task<IActionResult> Submit(string formId)
+        {
+            var form = await _contentManager.GetAsync(formId, VersionOptions.Published);
+
+            if(form == null)
+            {
+                return NotFound();
+            }
+
+            var formPart = form.As<AjaxForm>();
+
+            // Verify if the form is enabled
+            if (!formPart.Enabled.Value)
+            {
+                return NotFound();
+            }
+            var modelUpdater = _updateModelAccessor.ModelUpdater;
+            // bind form values to model state for if validation errors occur
+            foreach (var item in Request.Form)
+            {
+                modelUpdater.ModelState.SetModelValue(item.Key, item.Value, item.Value);
+            }
+
+            // form validation widgets
+            var flow = form.As<FlowPart>();
+            ValidateWidgets(flow.Widgets);
+
+            // form validation script
+            var script = form.As<AjaxFormScripts>();
+            var scriptingProvider = new AjaxFormMethodsProvider(form, modelUpdater);
+            if (!string.IsNullOrEmpty(script?.OnValidation?.Text))
+            {
+                _scriptingManager.EvaluateJs(script.OnValidation.Text, scriptingProvider);
+            }
+
+            var isValid = ModelState.ErrorCount == 0;
+
+            if (!isValid)
+            {
+                var model = await _contentItemDisplayManager.BuildDisplayAsync(form, modelUpdater);
+                var formHtml = await this.RenderViewAsync("Display", model, true);
+                return Json(new { validationError = true, html = formHtml });
+            }
+
+            if (!string.IsNullOrEmpty(script?.OnSubmitted?.Text))
+            {
+                _scriptingManager.EvaluateJs(script.OnSubmitted.Text, scriptingProvider);
+            }
+
+            // _workflow manager is null if workflow feature is not enabled
+            if (_workflowManager != null)
```

Comment:
This won't be null if the feature is not enabled, the ctor would just fail.

Reply:
This did worked, as the ctor takes a optional parameter. I was not sure how to do this otherwise.